webrtc/stun/stun_connection.nim

```diff
@@ -0,0 +1,116 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import chronos, chronicles
+import stun_protocol, ../[udp_connection, errors]
+import ../udp_connection, stun_protocol
+
+logScope:
+  topics = "webrtc stun stun_connection"
+
+# TODO:
+# - Need to implement ICE-CONTROLL(ED|ING) for browser to browser (not critical)
+
+type
+  StunConn* = ref object
+    conn*: UdpConn
+    laddr*: TransportAddress
+    raddr*: TransportAddress
+    dataRecv*: AsyncQueue[seq[byte]]
+    stunMsgs*: AsyncQueue[seq[byte]]
+    handlesFut*: Future[void]
+    closeEvent: AsyncEvent
+    closed*: bool
```

Comment:
Could we please have some docs for these fields?

Reply:
Done

Reply:
Done